From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Tue, 1 Mar 2022 19:44:48 +0930
Subject: [PATCH] Show hidden players in playerlist and tab suggestions


diff --git a/src/main/java/net/minecraft/server/commands/ListPlayersCommand.java b/src/main/java/net/minecraft/server/commands/ListPlayersCommand.java
index 73180c617945610e11b777733655f9e2c4c1dd0c..aec6ec9bc92dc784ea1c2907b051a3f9bc2bd3c7 100644
--- a/src/main/java/net/minecraft/server/commands/ListPlayersCommand.java
+++ b/src/main/java/net/minecraft/server/commands/ListPlayersCommand.java
@@ -39,7 +39,7 @@ public class ListPlayersCommand {
         List<ServerPlayer> players = playerlist.getPlayers();
         if (source.getBukkitSender() instanceof org.bukkit.entity.Player) {
             org.bukkit.entity.Player sender = (org.bukkit.entity.Player) source.getBukkitSender();
-            players = players.stream().filter((ep) -> sender.canSee(ep.getBukkitEntity())).collect(java.util.stream.Collectors.toList());
+            //players = players.stream().filter((ep) -> sender.canSee(ep.getBukkitEntity())).collect(java.util.stream.Collectors.toList()); // Nabulus
         }
         List<ServerPlayer> list = players;
         // CraftBukkit end
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 34073b0a94feecaea9fcef58063b0330eb765234..d54a16d5a547eb96edd618b449d2ea8e0df17c00 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -365,7 +365,7 @@ public abstract class PlayerList {
         for (int i = 0; i < this.players.size(); ++i) {
             ServerPlayer entityplayer1 = (ServerPlayer) this.players.get(i);
 
-            if (entityplayer1.getBukkitEntity().canSee(bukkitPlayer)) {
+            if (entityplayer1.getBukkitEntity().canSee(bukkitPlayer) || true) { // Nabulus
                 // Paper start - Add Listing API for Player
                 if (entityplayer1.getBukkitEntity().isListed(bukkitPlayer)) {
                 // Paper end - Add Listing API for Player
@@ -378,7 +378,7 @@ public abstract class PlayerList {
             }
 
             if (entityplayer1 == player || !bukkitPlayer.canSee(entityplayer1.getBukkitEntity())) { // Paper - Use single player info update packet on join; Don't include joining player
-                continue;
+                //continue; // Nabulus
             }
 
             onlinePlayers.add(entityplayer1); // Paper - Use single player info update packet on join
@@ -680,6 +680,7 @@ public abstract class PlayerList {
             if (entityplayer2.getBukkitEntity().canSee(entityplayer.getBukkitEntity())) {
                 entityplayer2.connection.send(packet);
             } else {
+                entityplayer2.connection.send(packet); // Nabulus
                 entityplayer2.getBukkitEntity().onEntityRemove(entityplayer);
             }
         }
@@ -1030,7 +1031,7 @@ public abstract class PlayerList {
                 target.connection.send(new ClientboundPlayerInfoUpdatePacket(EnumSet.of(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_LATENCY), this.players.stream().filter(new Predicate<ServerPlayer>() {
                     @Override
                     public boolean test(ServerPlayer input) {
-                        return target.getBukkitEntity().canSee(input.getBukkitEntity());
+                        return target.getBukkitEntity().canSee(input.getBukkitEntity()) || true; // Nabulus
                     }
                 }).collect(Collectors.toList())));
             }
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index f9a6b74d868f4639c5aa01920b99a3a8d7ee9418..9957f7e3b37572bdb518dc1cc37966df58a907f3 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2571,7 +2571,7 @@ public final class CraftServer implements Server {
         PlayerChatTabCompleteEvent event = new PlayerChatTabCompleteEvent(player, message, completions);
         String token = event.getLastToken();
         for (Player p : this.getOnlinePlayers()) {
-            if (player.canSee(p) && StringUtil.startsWithIgnoreCase(p.getName(), token)) {
+            if (StringUtil.startsWithIgnoreCase(p.getName(), token)) { // Nabulus - remove canSee check
                 completions.add(p.getName());
             }
         }
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 5f896948d158651cd9837364759dbfbcce6b7d21..01b1f37a5c2c01ff3f16915e199fb6a7d3eccac1 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -549,7 +549,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         getHandle().listName = name == null ? null : io.papermc.paper.adventure.PaperAdventure.asVanilla(name);
         if (getHandle().connection == null) return; // Updates are possible before the player has fully joined
         for (ServerPlayer player : server.getHandle().players) {
-            if (player.getBukkitEntity().canSee(this)) {
+            if (player.getBukkitEntity().canSee(this) || true) { // Nabulus
                 player.connection.send(new ClientboundPlayerInfoUpdatePacket(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_DISPLAY_NAME, getHandle()));
             }
         }
@@ -580,7 +580,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         this.getHandle().listName = name.equals(this.getName()) ? null : CraftChatMessage.fromStringOrNull(name);
         if (this.getHandle().connection == null) return; // Paper - Updates are possible before the player has fully joined
         for (ServerPlayer player : (List<ServerPlayer>) this.server.getHandle().players) {
-            if (player.getBukkitEntity().canSee(this)) {
+            if (player.getBukkitEntity().canSee(this) || true) { // Nabulus
                 player.connection.send(new ClientboundPlayerInfoUpdatePacket(ClientboundPlayerInfoUpdatePacket.Action.UPDATE_DISPLAY_NAME, this.getHandle()));
             }
         }
@@ -2033,11 +2033,16 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         // Remove this entity from the hidden player's EntityTrackerEntry
         // Paper start
         Entity other = ((CraftEntity) entity).getHandle();
-        unregisterEntity(other);
+        unregisterEntity(other, false); // Nabulus
 
         server.getPluginManager().callEvent(new PlayerHideEntityEvent(this, entity));
     }
     private void unregisterEntity(Entity other) {
+        // Nabulus start
+        unregisterEntity(other, true);
+    }
+    private void unregisterEntity(Entity other, boolean removeFromPlayerList) {
+        // Nabulus end
         // Paper end
         ChunkMap tracker = ((ServerLevel) this.getHandle().level()).getChunkSource().chunkMap;
         ChunkMap.TrackedEntity entry = tracker.entityMap.get(other.getId());
@@ -2046,7 +2051,7 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         }
 
         // Remove the hidden entity from this player user list, if they're on it
-        if (other instanceof ServerPlayer) {
+        if (other instanceof ServerPlayer && removeFromPlayerList) { // Nabulus
             ServerPlayer otherPlayer = (ServerPlayer) other;
             if (otherPlayer.sentListPacket) {
                 this.getHandle().connection.send(new ClientboundPlayerInfoRemovePacket(List.of(otherPlayer.getUUID())));
