From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Sat, 16 Dec 2023 15:18:46 +0930
Subject: [PATCH] Don't show smoke for silent fireworks


diff --git a/src/main/java/net/minecraft/world/entity/projectile/FireworkRocketEntity.java b/src/main/java/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
index 3df4005b6ddc075e5922c1e96e4ceb21d8b8ea43..825d8652c6ccd75a178b691334b663b54411c336 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/FireworkRocketEntity.java
@@ -176,7 +176,7 @@ public class FireworkRocketEntity extends Projectile implements ItemSupplier {
         }
 
         ++this.life;
-        if (this.level().isClientSide && this.life % 2 < 2) {
+        if (this.level().isClientSide && this.life % 2 < 2 && !this.isSilent()) { // Nabulus - don't show firework smoke for silent fireworks
             this.level().addParticle(ParticleTypes.FIREWORK, this.getX(), this.getY(), this.getZ(), this.random.nextGaussian() * 0.05D, -this.getDeltaMovement().y * 0.5D, this.random.nextGaussian() * 0.05D);
         }
 
