From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: froobynooby <froobynooby@froobworld.com>
Date: Mon, 11 Mar 2024 18:40:31 +0930
Subject: [PATCH] Allow players to hide wither boss bar


diff --git a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
index de2471cfa96a23944f229f33ffdff88b6b7756e4..dd3c894d766b88ac288cc4ea35c21efe6eb4f2d7 100644
--- a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -90,6 +90,21 @@ public class WitherBoss extends Monster implements PowerableMob, RangedAttackMob
 
     public void setCanTravelThroughPortals(boolean canPortal) { this.canPortal = canPortal; }
     // Paper end
+    // Nabulus start
+    public boolean hideBossBar = false;
+    private void updateBossBar() {
+        Component customName = getCustomName();
+        if (customName != null) {
+            String collapsedName = customName.tryCollapseToString();
+            hideBossBar = collapsedName != null && collapsedName.equalsIgnoreCase("nobossbar");
+        } else {
+            hideBossBar = false;
+        }
+        if (bossEvent != null && hideBossBar) {
+            bossEvent.removeAllPlayers();
+        }
+    }
+    // Nabulus end
 
     public WitherBoss(EntityType<? extends WitherBoss> type, Level world) {
         super(type, world);
@@ -148,6 +163,7 @@ public class WitherBoss extends Monster implements PowerableMob, RangedAttackMob
     @Override
     public void setCustomName(@Nullable Component name) {
         super.setCustomName(name);
+        updateBossBar(); // Nabulus
         this.bossEvent.setName(this.getDisplayName());
     }
 
@@ -422,7 +438,7 @@ public class WitherBoss extends Monster implements PowerableMob, RangedAttackMob
     @Override
     public void startSeenByPlayer(ServerPlayer player) {
         super.startSeenByPlayer(player);
-        this.bossEvent.addPlayer(player);
+        if (!hideBossBar) this.bossEvent.addPlayer(player); // Nabulus
     }
 
     @Override
